### PR TITLE
[WIP] Checking if build is true before prompting or initializing

### LINKED ISSUE #693: `func deploy` should only prompt build questions if `build` argument is set to `true`
Currently, even if you disable the build option for `func deploy` it still prompt questions related to the build phase. 

https://github.com/knative-sandbox/kn-plugin-func/blob/c2e1b769cc0d80082e6ef1be9ad2785ba8db4550/cmd/deploy.go#L114

I think this is just a bug.

### PR DESCRIPTION
**Pull request by salaboy**
# Changes
- :bug: Checks the build configuration argument before prompting build questions or trying to check if the func project has been initialized. 

/kind bug

Fixes #693

### CODE REVIEW

**Comment by salaboy** [comment]
@shinigambit This is just a way of doing this.. we need unit tests for this one .. and see what other people thinks about this change.

File: cmd/deploy.go [hunk 1]
```diff
@@ -111,27 +111,32 @@ func runDeploy(cmd *cobra.Command, _ []string, clientFn deployClientFn) (err err
...
+		function.Envs, err = mergeEnvs(function.Envs, config.EnvToUpdate, config.EnvToRemove)
+		if err != nil {
+			return
+		}
 	}
 
-	// Check if the Function has been initialized
-	if !function.Initialized() {
-		return fmt.Errorf("the given path '%v' does not contain an initialized function. Please create one at this path before deploying", config.Path)
+	if config.Build {
```
Comment (lkingland) [inline cmd/deploy.go:126]: This is a good idea, but I think it should actually be part of the `Prompt` method.  For two reasons, first, this causes the user to never be prompted for other attributes, such as Namespace, which are pertinent whether or not build is enabled.  Second, the purpose of this `if` statement is to adjust how `Prompt` is implemented, so it should be part of Prompt's implementation.

The way this is can be handled is by splitting Prompt into multiple sets of queries (you can see this in action in `invoke.go` which only asks for `Data` if `File` was not specified.

**Review by lkingland — CHANGES_REQUESTED** [review]
Perhaps consider placing the logic in `.Prompt()` to avoid skipping prompts for other attributes.

**Comment by lkingland** [comment]
The referenced issue was fixed in #1071

**Pull request closed without merging**